### PR TITLE
fix: Fix cpu-only-py3-base job

### PR DESCRIPTION
**Pull request by yinggeh**
#### What does the PR do?
Update CMakeLists.txt to make sure CUDA libraries are not included in the build process when TRITON_ENABLE_GPU=OFF.

#### Checklist
- [x] PR title reflects the change and is of format `<commit_type>: <Title>`
- [x] Changes are described in the pull request.
- [x] Related issues are referenced.
- [x] Populated [github labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) field
- [x] Added [test plan](#test-plan) and verified test passes.
- [x] Verified that the PR passes existing CI.
- [x] Verified copyright is correct on all changed files.
- [x] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [x] All template sections are filled out.
- [ ] Optional: Additional screenshots for behavior/output changes with before/after.

#### Commit Type:
- [x] fix

#### Related PRs:
N/A

#### Where should the reviewer start?
N/A

#### Test plan:
N/A

- CI Pipeline ID:
16347752

#### Caveats:
N/A

#### Background
N/A

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)
N/A

### CODE REVIEW

File: src/test/CMakeLists.txt
Comment (fpetrini15) [inline src/test/CMakeLists.txt:669]: My preference would be to merge this if/else with the if/else above. Doing so would also make it more consistent with the rest of the file.

Comment (yinggeh) [reply src/test/CMakeLists.txt:669]: My test cases are partially run in CPU.

Comment (fpetrini15) [reply src/test/CMakeLists.txt:669]: Right, but I think it can be condensed into a single if/else block for clarify. I'll leave it up to you.

**Review by fpetrini15 — APPROVED** [review]
Optional comment. Otherwise LGTM, thanks for the quick fix!

File: src/test/CMakeLists.txt
Comment (yinggeh) [reply src/test/CMakeLists.txt]: I agree. Fixed.

**Review by fpetrini15 — APPROVED** [review]
LGTM, pending passing pipeline. Great work!

**Pull request merged**
